docs/whatsnew/7.0.rst

```diff
@@ -288,6 +289,29 @@ New ``Lorentz2D`` model
 A new 2D Lorentzian model has been added to the ``astropy.modeling``
 package.
 
+.. _whatsnew-7.0-masked-coordinates:
+
+Support for masks in ``coordinates``
+====================================
+
+Initial support for masks has been implemented for |SkyCoord|, as well as for
+coordinate frames and coordinate representations.
+
+Like more generally for |Masked| instances, one can access the mask of using
+``.mask``, check whether the underlying data are masked using ``masked``,
+access underlying unmasked data using ``.unmasked``, and get a new instance
+with masked data replaced with a fill value using the ``filled()`` method.
+
+By default, the mask combines all masks of the underlying representation
+components, including those on possible differentials like radial velocity
+and proper motion. It is possible to make other choices using the new
+``.get_mask()`` method. For instance, ``sc.get_mask("ra", "dec")`` would
+get the combined masks of just the angular components, which may be useful
+if dealing with, e.g., a catalogue of stars where distances are available
+only for some of the objects.
```

Comment:
It should also be mentioned that the frame attributes can be masked too and accessed with `get_mask()`.

Reply:
Arguably too much for the what's-new, but would definitely be good to add to the actual docs.